client/src/components/RatingsAndReviews/RatingsAndReviews.jsx

```diff
@@ -5,17 +5,121 @@ import ReviewsList from './ReviewsList/ReviewsList.jsx';
 import WritingReview from './WritingReview/WritingReview.jsx';
 import RatingBreakdown from './RatingBreakdown/RatingBreakdown.jsx';
 import ProductBreakdown from './ProductBreakdown/ProductBreakdown.jsx';
+import axios from 'axios';
 
-const RatingsAndReviews = () => {
-  return (
-    <div className='ratings-and-reviews'>
-      <Sorting />
-      <ReviewsList />
-      <WritingReview />
-      <RatingBreakdown />
-      <ProductBreakdown />
-    </div>
-  );
-};
+class RatingsAndReviews extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      reviews: [],
+      meta: {},
+      showMoreReviewsButton: true,
+      averageRating: 0,
+      totalNumberRating: 0,
+      percentRecommend: 0,
+      ratingArray: [],
+      numStarReviewsToRender: []
+    };
+    this.calcRating = this.calcRating.bind(this);
+    this.filterReviews = this.filterReviews.bind(this);
+  }
 
+  componentDidMount() {
+    axios.get(`/reviews/?product_id=${this.props.productId}`)
+      .then(response => {
+        console.log('response from axios get:', response.data.results);
+        this.setState({ reviews: response.data.results }, () => {
+          if (this.state.reviews.length <= 2) {
+            this.setState({ showMoreReviewsButton: false });
+          }
+        });
+      })
+      .catch(err => {
+        console.log('error');
+      });
+
+    axios.get(`/reviews/meta/?product_id=${this.props.productId}`)
+      .then(response => {
+        console.log('response from axios get meta:', response.data);
+        this.setState({ meta: response.data }, () => this.calcRating.bind(this)(this.state.meta));
```

Comment:
`this.calcRating` is already binded to this at the constructor. I wonder if you need to bind it again here.